src/ffi2/datetime.rs

```diff
@@ -0,0 +1,354 @@
+use ffi2::object::*;
+use ffi2::pycapsule::PyCapsule_Import;
+use ffi2::pyport::Py_hash_t;
+use std::ffi::CString;
+use std::option::Option;
+use std::os::raw::{c_char, c_int, c_uchar};
+
+#[cfg_attr(windows, link(name = "pythonXY"))]
+extern "C" {
+    pub static mut PyDateTime_DateType: PyTypeObject;
+    pub static mut PyDateTime_TimeType: PyTypeObject;
+    pub static mut PyDateTime_DateTimeType: PyTypeObject;
+
+    pub static mut PyDateTime_DeltaType: PyTypeObject;
+    pub static mut PyDateTime_TZInfoType: PyTypeObject;
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
```

Comment:
No, I'm sure I copy-pasted this. I'm not entirely sure what the consequences are of deriving `Copy` and `Clone`, but I imagine that if you `memcpy` `PyDateTime_Date` it would be fine, but the other two hold a reference to a `PyObject`, and so there are reference counting implications of making a `memcpy` without discarding the original object.

I'll remove at least `Copy`, but I dunno if `Clone` should also get the same treatment.

Reply:
By the way, I figured out the reason why it was `#[derive(Debug, Copy, Clone)]` - it's because I used `rust-bindgen` to generate the initial bindings.